### ISSUE TITLE
Moving `~/.opam` and issues with soft linking it

### ISSUE THREAD

**Issue by SaswatPadhi** [ocaml/opam#3513]
Hello,

Due to space issues, I have to move my (> 8 GB) `~/.opam` directory to another drive. Does `opam` provide a way to change this directory?

I couldn't find a way to do that, so I moved the `.opam` directory to the other drive and soft linked it to `~/.opam`. The existing packages all work fine without any issues, but when I install a new package, I see errors like:

```
[ERROR] The compilation of user-setup failed at
        "/home/user/.opam/opam-init/hooks/sandbox.sh build make".

#=== ERROR while compiling user-setup.0.7 =====================================#
# context              2.0.0 | linux/x86_64 | ocaml-variants.4.07.0+flambda | https://opam.ocaml.org/2.0#a6bb6a42
# path                 ~/.opam/4.07.0+flambda/.opam-switch/build/user-setup.0.7
# command              ~/.opam/opam-init/hooks/sandbox.sh build make
# exit-code            1
# env-file             ~/.opam/log/user-setup-6663-05296d.env
# output-file          ~/.opam/log/user-setup-6663-05296d.out
### output ###
# Can't mkdir /home/user/.opam/4.07.0+flambda: No such file or directory
```

Here is the `opam config report` generated on my machine:
```
# opam config report
# opam-version      2.0.0 
# self-upgrade      no
# system            arch=x86_64 os=linux os-distribution=debian os-version=9.5
# solver            builtin-mccs+glpk
# install-criteria  -removed,-count[version-lag,request],-count[version-lag,changed],-changed
# upgrade-criteria  -removed,-count[version-lag,solution],-new
# jobs              5
# repositories      1 (http) (default repo at a6bb6a42)
# pinned            0
# current-switch    4.07.0+flambda
```

**Comment by Khady** [comment]
I think you can use the `OPAMROOT` env variable to indicate the path to your `.opam` directory.

**Comment by AltGr** [comment]
> Interestingly, the line that opam added to my ~/.bashrc during step 2, didn't seem to work!

Ah! Indeed, I guess we have all been using `opam init --root xxx` to setup a _secondary_ root, either for testing purposes or for multiple opam versions: in which case that's what we want, but we obviously don't choose the option to update the init scripts anyway. Good call, thanks for reporting!

**Comment by rjbou** [comment]
opam now resolves opam root (PR #3681). 
Thanks for the report!
